src/taco-cli/cli/remote.ts

```diff
@@ -208,11 +208,11 @@ class Remote extends commands.TacoCommandBase {
     }
 
     private static queryUserForRemoteConfig(): Q.Promise<{ host: string; port: number; pin: number }> {
-        var hostPromise = Q.defer<{ host: string }>();
-        var portPromise = Q.defer<{ host: string; port: number }>();
-        var pinPromise = Q.defer<{ host: string; port: number; pin: number }>();
+        var hostPromise: Q.Deferred<{ host: string }> = Q.defer<{ host: string }>();
+        var portPromise: Q.Deferred<{ host: string; port: number }> = Q.defer<{ host: string; port: number }>();
+        var pinPromise: Q.Deferred<{ host: string; port: number; pin: number }> = Q.defer<{ host: string; port: number; pin: number }>();
 
-        var cliSession = Remote.cliSession ? Remote.cliSession : readline.createInterface({ input: process.stdin, output: process.stdout });
+        var cliSession: IcliSession = Remote.cliSession ? Remote.cliSession : readline.createInterface({ input: process.stdin, output: process.stdout });
```

Comment:
Not part of this change but per our coding guidelines, the name of this interface should be "ICliSession" not "IcliSession"